### PR TITLE
Dovecot: Do not include empty sieve_extensions and sieve_global_extensions

### PR DESCRIPTION
**Pull request by exi**
## Description of changes

Setting them to empty string will disable the default behaviour, leading to missing extensions.

@2xsaiko Thanks for adding proper imapsieve support, it let me remove a bunch of custom configs!

### CODE REVIEW

**Comment by 2xsaiko** [comment]
This should already be covered by https://github.com/NixOS/nixpkgs/pull/281001, but thanks! And sorry for the trouble 😄

**Comment by 2xsaiko** [comment]
Though, this might be faster to merge first since it's just a small change.

**Review by dblsaiko — APPROVED** [review]

File: nixos/modules/services/mail/dovecot.nix [hunk 1]
```diff
@@ -119,9 +119,10 @@ let
     ''
       plugin {
         sieve_plugins = ${concatStringsSep " " cfg.sieve.plugins}
-        sieve_extensions = ${concatStringsSep " " (map (el: "+${el}") cfg.sieve.extensions)}
-        sieve_global_extensions = ${concatStringsSep " " (map (el: "+${el}") cfg.sieve.globalExtensions)}
     ''
+    (optionalString (cfg.sieve.extensions != []) ''sieve_extensions = ${concatStringsSep " " (map (el: "+${el}") cfg.sieve.extensions)}'')
+    (optionalString (cfg.sieve.globalExtensions != []) ''sieve_global_extensions = ${concatStringsSep " " (map (el: "+${el}") cfg.sieve.globalExtensions)}'')
```
Comment (h7x4) [inline nixos/modules/services/mail/dovecot.nix:124]: So there's this magic function...

```suggestion
    (optionalString (cfg.sieve.extensions != []) ''sieve_extensions = ${concatMapStringsSep " " (el: "+${el}") cfg.sieve.extensions}'')
    (optionalString (cfg.sieve.globalExtensions != []) ''sieve_global_extensions = ${concatMapStringsSep " " (el: "+${el}") cfg.sieve.globalExtensions}'')
```

**Review by dblsaiko — APPROVED** [review]

**Review by h7x4 — APPROVED** [review]

**Review by kevincox — APPROVED** [review]

**Comment by kevincox** [comment]
Anything blocking merging?

**Comment by 2xsaiko** [comment]
Not from me!

**Comment by dotlambda** [comment]
This broke dovecot for me:
```
doveconf: Fatal: Error in configuration file /etc/dovecot/dovecot.conf line 75: Section not allowed here
```
because `/etc/dovecot/dovecot.conf` contains these lines:
```
plugin {
  sieve_plugins = 
sieve_extensions = +fileinto}
#Extra Config


service imap-login {
```
and my configuration uses [Simple NixOS mailserver](https://gitlab.com/simple-nixos-mailserver/nixos-mailserver) but also contains
```nix
services.dovecot2.sieve.extensions = [ "fileinto" ];
```

File: nixos/modules/services/mail/dovecot.nix [hunk 2]
```diff
@@ -119,9 +119,10 @@ let
     ''
       plugin {
         sieve_plugins = ${concatStringsSep " " cfg.sieve.plugins}
-        sieve_extensions = ${concatStringsSep " " (map (el: "+${el}") cfg.sieve.extensions)}
-        sieve_global_extensions = ${concatStringsSep " " (map (el: "+${el}") cfg.sieve.globalExtensions)}
     ''
+    (optionalString (cfg.sieve.extensions != []) ''sieve_extensions = ${concatMapStringsSep " " (el: "+${el}") cfg.sieve.extensions}'')
+    (optionalString (cfg.sieve.globalExtensions != []) ''sieve_global_extensions = ${concatMapStringsSep " " (el: "+${el}") cfg.sieve.globalExtensions}'')
```
Comment (2xsaiko) [inline nixos/modules/services/mail/dovecot.nix:124]:
```suggestion
    (optionalString (cfg.sieve.extensions != []) ''
      sieve_extensions = ${concatMapStringsSep " " (el: "+${el}") cfg.sieve.extensions}
    '')
    (optionalString (cfg.sieve.globalExtensions != []) ''
      sieve_global_extensions = ${concatMapStringsSep " " (el: "+${el}") cfg.sieve.globalExtensions}
    '')
```

**Review by dblsaiko — COMMENTED** [review]
Oh right, like this it's no longer adding new lines after the two lines. I completely missed that, sorry!

**Pull request merged**